src/main/java/Task.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Represents a Task, which is an abstract class that can be used to represent
+ * different types of tasks.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Creates a Task object with the given description.
+     *
+     * @param description the description of the task
+     * @throws PandaBotEmptyTaskDescriptionException If the description given is empty
+     */
+    public Task(String description) throws PandaBotEmptyTaskDescriptionException {
+        this.description = description.strip(); // removes starting and ending white spaces
```

Comment:
Redundant comment, there's no need to explain what strip does (it has its own javadocs).